### ISSUE TITLE
[Bug] Producer fails to send message size equals to maxMessageSize in topic policy setting.

### ISSUE THREAD

**Issue by Jason918** [apache/pulsar#12958]
**Describe the bug**
See the following unit test code to reproduce this issue.

The root cause of this is that in topic policy setting, it limits the size of message **header and payload**. 
See https://github.com/apache/pulsar/blob/6b8ebbd288085dc5a93a9e92706eb5f5ad397cb8/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L413

**To Reproduce**
Add this unit test code in org.apache.pulsar.broker.admin.TopicPoliciesTest
```
    @Test
    public void testTopicMaxMessageSizeThreshHold() throws Exception {
        @Cleanup
        Producer<byte[]> p = pulsarClient.newProducer().topic(testTopic).create();
        admin.topicPolicies().setMaxMessageSize(testTopic,1000);
        Awaitility.await().until(() -> pulsar.getTopicPoliciesService().getTopicPolicies(TopicName.get(testTopic)) != null);
        p.send(new byte[1000]);
    }
```

Got the following error:

> org.apache.pulsar.client.api.PulsarClientException$NotAllowedException: The size of the message which is produced by producer test-0-2 to the topic persistent://my-tenant/my-namespace/test-set-backlog-quota-partition-0 is not allowed

**Expected behavior**
Message should produce success.

**Screenshots**
NA

**Desktop (please complete the following information):**
 - OS: macOS

**Additional context**
NA

**Comment by Jason918** [comment]
I think we can add new field "payload size" in CommandSend API. 
And make topic policy only checks the payload size.
What do you think? @codelipenghui @315157973 @eolivelli

**Comment by Jason918** [comment]
This will be fixed by PIP-132 in #13591
Close this issue.
